### PR TITLE
Adds cli functionality

### PR DESCRIPTION
**Pull request by Nino-K**

### CODE REVIEW

File: pkg/commands/start.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+func Start(address string, udpLocalPort, tcpLocalPort int, IPv6 bool, hosts map[string]string) error {
+	//TODO: seperate the handler from start
+	srv, err := dns.Start(address, udpLocalPort, tcpLocalPort, IPv6, hosts)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Started srv %+v", srv)
```
Comment (mook-as) [inline pkg/commands/start.go:31]: Shouldn't there be a `defer srv.Shutdown()` somewhere?

**Review by mook-as — CHANGES_REQUESTED** [review]
Why do we have a subcommand here at all? Do we have plans for other commands? (If yes, where?)

File: pkg/commands/start.go (hunk 1 above)
Comment (Nino-K) [reply pkg/commands/start.go:31]: 👍  I will be adding more enhancements in the upcoming PRs, this PR us just adding the cmd stuff. There are still some changes that I will need to make in `pkg/dns`

**Comment by Nino-K** [comment]
> Why do we have a subcommand here at all? Do we have plans for other commands? (If yes, where?)

The `command/start.go` tends to be a wrapper around the `pkg/dns` thing. Maybe we will be adding other verbs in the future e.g `host-resolver flush`, `reset` etc, maybe not 🤷 . But for now, I needed a place to add other stuff, since we will be pushing all our changes for `pkg/dns` to the upstream and we will not have that forked locally. For now, it follows the `PROCESS VERB` logic, however, I'm open to suggestions for renaming `commands` to something that makes more sense.

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
+Copyright © 2022 SUSE LLC
+1
```
Comment (jandubois) [inline cmd/root.go]: What is this `1` here?

Also, do we really need this big license blob in every file? I don't see it in the Rancher Desktop Typescript sources, so why is this needed in the Go files?

Comment (Nino-K) [reply cmd/root.go]: Not sure how did that `1` got in there, probably from where I copied it from(now I need to figure out where I copied it from 😂 ). General practice for open-source software is to include the header license in all the files. We should probably make that decision to either include everywhere or what is the plan. Because I have seen the random appearance of license in file headers in `rancher-desktop`.

Comment (mook-as) [reply cmd/root.go]: Yes, we're _supposed_ to have those for Rancher Desktop TypeScript (and everything else) files too; we just haven't been doing it properly…

**Review by mook-as — APPROVED** [review]
LGTM.

We should start considering having tests.

**Comment by Nino-K** [comment]
> LGTM.
> 
> We should start considering having tests.

For sure, I'm adding them as we speak. Next PRs will have them.
